openedx/core/lib/api/authentication.py

```diff
@@ -65,19 +76,53 @@ class override.
     This class can be used for an OAuth2-accessible endpoint that allows users to access
     that endpoint without having their email verified.  For example, this is used
     for mobile endpoints.
-
     """
+
+    def authenticate(self, *args, **kwargs):
+        """
+        Returns two-tuple of (user, token) if access token authentication
+        succeeds, raises an AuthenticationFailed (HTTP 401) if authentication
+        fails or None if the user did not try to authenticate using an access
+        token.
+
+        Overrides base class implementation to return edX-style error
+        responses.
+        """
+
+        try:
+            return super(OAuth2AuthenticationAllowInactiveUser, self).authenticate(*args, **kwargs)
+        except AuthenticationFailed:
```

Comment:
Why was it necessary to catch our own exception here?
Regardless, since you're not handling that error here specifically, why not just let it pass through without catching and re-raising?